tasks/ollama-sentiment-analysis-fn/resources/function/function.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, Optional
+
+import time
+import logging
+import subprocess
+
+from coretex import functions
+from ollama import ResponseError
+
+import ollama
+import requests
+
+
+OLLAMA_SERVER_URL = "http://127.0.0.1:11434"
+TEXT_PROMPT_PREFIX = "Grade the sentiment of the following text: "
+SYSTEM_PROMPT = [
+    {
+        "role": "user",
+        "content": "You are a sentiment analysis bot. You will recieve a text which you will grade as one of five sentiment categories ([very positive], [positive], [neutral], [negative], [very negative]). Your respose will follow this format \"Sentiment: positive\". Is this understood?"
+    },
+    {
+        "role": "assistant",
+        "content": "Understood. All following responses will follow format \"Sentiment: category\""
+    },
+    {
+        "role": "user",
+        "content": f"{TEXT_PROMPT_PREFIX}\"Absolutely in love with this new product! The quality is unbelievable, and it's so advanced. Highly recommend to any humans out there!\""
+    },
+    {
+        "role": "assistant",
+        "content": "Sentiment: very positive"
+    }
+]
+
+
+def isOllamaInstalled() -> bool:
+    try:
+        subprocess.run(["ollama", "--version"], check = True, stdout = subprocess.PIPE, stderr = subprocess.PIPE)
+        return True
+    except (subprocess.CalledProcessError, FileNotFoundError):
+        return False
+
+
+def installOllama() -> None:
+    try:
+        subprocess.run("curl https://ollama.ai/install.sh | sh", shell = True, check = True)
+        logging.info(">> [OSentimentAnalysis] Ollama installation was successful")
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError(f">> [OSentimentAnalysis] An error occurred during Ollama installation: {e}")
+
+
+def checkOllamaServer() -> Optional[bool]:
+    try:
+        response = requests.get(OLLAMA_SERVER_URL)
+        return response.status_code == 200
+    except requests.ConnectionError:
+        return False
+
+
+def launchOllamaServer() -> Optional[subprocess.Popen[bytes]]:
+    if not isOllamaInstalled():
+        installOllama()
+
+    if checkOllamaServer():
+        return None
+
+    logging.info(">> [OSentimentAnalysis] Launching ollama sever")
+    return subprocess.Popen(["ollama", "serve"])
+
+
+def pullModel(attempt: Optional[int] = None) -> None:
```

Comment:
Update to be timeout based on seconds not on number of attempts. Wait for 1 min max before raising error